### DISCUSSION TITLE
CLI for FybrikModule Deploy & Taxonomy Updates

### DISCUSSION THREAD

**Discussion by simanadler** [fybrik/fybrik#1122] [Ideas]
When creating, updating or deleting a FybrikModule CRD deployed to the fybrik control plane the IT admin is in essence changing the capabilities and actions available for use by the data governance policies and the IT config policies, and ultimately the data plane.

Currently when one updates a module the relevant [module taxonomies ](https://github.com/fybrik/fybrik/tree/master/samples/taxonomy/example/module)also need to be manually updated before deploying the FybrikModule CRD.

Proposal: Create a command line interface (CLI) for deploying/updating/deleting modules, which will
(1) Update the taxonomy if necessary
(2) Use the relevant kubernetes command to deploy the FybrikModule CRD.

@roee88 @roytman @Mohammad-nassar10 @rohithdv Your thoughts?

**Comment by roytman** [comment]
@simanadler , sorry, but I want to be sure, when you are talking about `creating, updating or deleting a FybrikModule CRD`, are you talking about Custom Resource  `Definitions` (CRDs) or Custom Resources (CRs). 
We can create this CLI or extend kubectl. 
The problem with proprietary CLIs is that users should use a lot of different CLIs and therefore they do not always like it. 
Let's gather here proposed scenarios: command A will do a,b,c; command B ... and continue this discussion.
